conformance/lang/expressions/let-expr/let_expr.balt

```diff
@@ -0,0 +1,431 @@
+    string[] children = [];
+};
+
+type Student record {
+    int id;
+    string name?;
+};
+
+function init() {
+    io:println(let Family family1 = {spouse: "Ann", noOfChildren: 0}, Person emp1 = {family: family1},
+        Person[] emps = [emp1], Department dpt = {dptName: "HR", employees: emps} in dpt.employees[0].family);
+    // @output {"spouse":"Ann","noOfChildren":0,"children":[]}
+    io:println(let Family family1 = {spouse: "Ann", noOfChildren: 0}, Person emp1 = {family: family1}
+        in emp1.entries());
+    // @output {"name":["name","default name"],"family":["family",{"spouse":"Ann","noOfChildren":0,"children":[]}]}
+    io:println(let Student s = {id : 1, name : "Andrew"} in s.removeIfHasKey("name")); // @output Andrew
+    io:println(() == let Student s = {id : 1, name : "Andrew"} in s.removeIfHasKey("firstName")); // @output true
+}
+
+Test-Case: output
+Description: Test let-expr, when type-descriptor is a table-type-descriptor which is a structured-type-descriptor.
+Labels: let-expr, record-type, table-constructor-expr
+
+type Foo record {
+    readonly map<string> m;
+    int age;
+};
+
+type Student record {|
+    readonly string name;
+    int id?;
+    Address address;
+|};
+
+type Address record {
+    string city;
+    string? country;
+};
+
+function init() {
+    io:println(let table<map<any>> tbl = table [{ id: 13 , name: "Emm", lname: "Dean" }] in tbl);
+    // @output [{"id":13,"name":"Emm","lname":"Dean"}]
+    io:println(() == let table<Foo> key(m) tbl = table [{m: {"AAA": "DDDD"}, age: 31 }]
+        in tbl.add({ m: {"CCC":"EEEE"}, age: 34 })); // @output true
+    io:println(let table<Student> key(name) tbl = table [{name: "Amy", id: 1234, address:
+        {"street": "Main Street", "city": "Colombo", "country": ()}}] in tbl["Amy"]["address"]["country"] == ());
+    // @output true
+}
+
+Test-Case: output
+Description: Test let-expr with query expressions.
+Labels: let-expr, record-type, array-type, list-constructor-expr
+
+type Department record {|
+   int id;
+   string name;
+|};
+
+type Person record {|
+   int id;
+   string fname;
+   string lname;
+|};
+
+type DeptPerson record {|
+   string fname;
+   string lname;
+   string? dept;
+|};
+
+Person[] personList = [
+    {id: 1, fname: "Alex", lname: "George"}
+];
+
+Department[] deptList = [
+    {id: 1, name:"HR"}
+];
+
+function init() {
+    var l = let Person[] personList = [{id: 1, fname: "Sam", lname: "George"}],
+                   Department[] deptList = [{id: 1, name:"HR"}] in
+                   from var person in personList
+                        join var {id: deptId, name: deptName} in deptList
+                        on person.id equals deptId
+                        select {
+                            fname : person.fname,
+                            lname : person.lname,
+                            dept : deptName
+                        };
+    io:println(l); // @output [{"fname":"Sam","lname":"George","dept":"HR"}]
+}
+
```

Comment:
Will it be possible to add a test to cover?

> A let-expr is isolated if the expression following in is isolated; a reference in that expression to a variable bound by a let-var-decl will be treated as isolated if the expression initializing the variable in the let-var-decl is isolated.

Reply:
Added